### ISSUE TITLE
Messages of last Contact are repeatedly delivered on Startup

### ISSUE THREAD

**Issue by Catscratch** [jitsi/jitsi#341]
Hi,

I'm part of a group chat (XMPP) which Jitsi joins automatically on startup. But on every startup I get the last messages of the last user delivered again and again and Jitsi is showing it as new (window blinking etc.)

E.g. When the last user who (posted something) posted three messages than I get three toasts on startup (in Windows).

I also tried other messangers like Pidgin or Psi which doesn't do so.

Is there anything I have to configure? Or is it simply a bug?

I'm using version 2.11.5558. But it also happens in latest stable build.

Thanks for hints,
regards.

**Comment by damencho** [comment]
Hi, I remember working on similar issue long time ago and this should be handled, but to confirm this we need more info and logs. 
Messages like that should have a delayed extension (<delay xmlns='urn:xmpp:delay'), this can be checked in the pcap that is generated in the logs. 
So reproduce the issue and save the logs https://jitsi.org/faq#logs and send them to us.
Also which server are you using for xmpp?

**Comment by Catscratch** [comment]
Hi,

I'm not sure I can give you the logs because of some sensitive information in the messages. Really, sorry.

But I took a look. There indeed is a delay inside:
`<delay xmlns="urn:xmpp:delay" stamp="2017-03-10T08:14:42.083Z" from="***/XMPP Client"></delay>`

I got the message today. The delay points to a time 11 days ago.

Are there any information I can extract for you?

**Comment by damencho** [comment]
Nope, I'm good this needs more debugging. Currently I cannot remember where is the filtering made so we do not show this message twice, I need to spend more time on that, which I currently do not have.
Just more info on the subject if somebody starts looking into that:
- the server used and version (if possible)
- are you seeing this only for one2one chats, or it is for multichats?

**Comment by ibauersachs** [comment]
Btw, the .pcap logs don't contain the message content. You can verify that if open it in e.g. Wireshark.

**Comment by Catscratch** [comment]
Server: OpenFire 4.1.2
And only for groupchats.

**Comment by Catscratch** [comment]
We upgraded to latest stable of OpenFire (4.1.6). The problem still persists.
